spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -192,6 +194,14 @@ public Map<String, String> extraSnapshotMetadata() {
           }
         });
 
+    final Iterator<Tuple2<String, String>> it = sessionConf.getAll().iterator();
```

Comment:
Iceberg doesn't use `final` for local variables, since there isn't value in using it.